File: Parte3/ej3p3.m
% Vectores

% Escribir una función que reciba dos vectores y devuelva si son paralelos o no.

function paralelo = son_paralelos(v1, v2)
    if length(v1) ~= length(v2)
        error('Los vectores deben tener la misma longitud');
    end
    razones = v1 ./ v2;
    paralelo = all(razones == razones(1));
end

% Ejemplo de uso:
disp(son_paralelos([2, 4, 6], [1, 2, 3])); % Debería ser true

File: Parte3/ej3p3.c
//Vectores

// Escribir una función que reciba dos vectores y devuelva si son paralelos o no.

#include <stdio.h>

int son_paralelos(int v1[], int v2[], int n) {
    double razon = (double)v1[0] / v2[0];

    for (int i = 1; i < n; i++) {
        if ((double)v1[i] / v2[i] != razon) {
            return 0;
        }
    }
    return 1;
}

int main() {
    int v1[] = {2, 4, 6};
    int v2[] = {1, 2, 3};
    int n = 3;

    if (son_paralelos(v1, v2, n)) {
        printf("Los vectores son paralelos.\n");
    } else {
        printf("Los vectores NO son paralelos.\n");
    }
    return 0;
}
s